salt/config/__init__.py

```diff
@@ -1753,55 +1754,43 @@ def _gather_buffer_space():
 # <---- Salt Cloud Configuration Defaults ------------------------------------
 
 
-def _normalize_roots(file_roots):
+def _validate_roots(opts, prop_name):
     """
-    Normalize file or pillar roots.
+    If the roots option has a key that is None then we will log a warning and
+    use the defaults instead.
     """
-    for saltenv, dirs in file_roots.items():
-        normalized_saltenv = str(saltenv)
-        if normalized_saltenv != saltenv:
```

Comment:
I'm not seeing this if logic replicated anywhere. Am I possibly missing it?

Reply:
It is covered by line 1775, since the previous implementation iterated over the file roots and would pop and insert entries to it. The new version just creates an entirely new dynamic dictionary and therefore doesn't need to do this comparison and/or pop any entries in the roots.